File: clojure/sum-of-multiples/src/sum_of_multiples.clj
(ns sum-of-multiples)

(defn- divisible-by? [num div]
  (zero? (rem num div)))

(defn- divisible-by-some? [divs num]
  (some (partial divisible-by? num) divs))

(defn sum-of-multiples [divs max]
  (reduce + (filter (partial divisible-by-some? divs) (range max))))

File: go/sum-of-multiples/sum_of_multiples.go
package summultiples

func SumMultiples(limit int, divisors ...int) int {
	var sum int
	for n := 0; n < limit; n++ {
		for _, div := range divisors {
			if div != 0 && n%div == 0 {
				sum += n
				break
			}
		}
	}
	return sum
}
